src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/pipeline/JiraSendBuildInfoStep.java

```diff
@@ -100,9 +106,16 @@ public JiraSendBuildInfoStepExecution(
         @Override
         protected JiraSendInfoResponse run() throws Exception {
             final TaskListener taskListener = getContext().get(TaskListener.class);
-            final WorkflowRun build = getContext().get(WorkflowRun.class);
+            final WorkflowRun workflowRun = getContext().get(WorkflowRun.class);
+            final Optional<String> site = pickDefaultSite(step.getSite());
+
+            if (!site.isPresent()) {
+                workflowRun.setResult(Result.FAILURE);
```

Comment:
Our plugin is a build post-action, it can't change the status of a build. If we wanted to change the status of the build, we would need to implement a different extension point.

In my opinion, we should not fail a build if a default site can't be found. I think we should log a warning or error on the Output console in this case - this is how we handle other configuration issues at the moment.